internal/defence/limiter_test.go

```diff
@@ -0,0 +1,80 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package defence
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+
+	"storj.io/storj/internal/testcontext"
+)
+
+func TestLimiter(t *testing.T) {
+	ctx := testcontext.New(t)
+	key1 := "email1@example.com"
+	key2 := "email2@example.com"
```

Comment:
Using email as the key is a bad idea, without other checks. Also let's use a test domain for the tests.

Reply:
key could be anything. I could possible make a key not string but interface{}

Reply:
Using string is completely fine, it helps with memory usage.